ppa/archive/util.py

```diff
@@ -0,0 +1,136 @@
+'''
+
+Utility code related to :mod:`ppa.archives`
+
+'''
+
+from collections import OrderedDict
+from json.decoder import JSONDecodeError
+
+from ppa.archive import hathi
+from ppa.archive.models import DigitizedWork
+from ppa.archive.signals import IndexableSignalHandler
+from ppa.archive.solr import get_solr_connection
+
+
+class HathiImporter:
+    '''Logic for creating new :class:`~ppa.archive.models.DigitizedWork`
+    records from HathiTrust. For use in views and manage commands.'''
+
+    existing_ids = None
+
+    #: status - successfully imported record
+    SUCCESS = 1
+    #: status - skipped because already in the database
+    SKIPPED = 2
+
+    #: human-readable message to display for result status
+    status_message = {
+        SUCCESS: 'Success',
+        SKIPPED: 'Skipped; already in the database',
+        hathi.HathiItemNotFound: 'Error loading record; check that id is valid.',
+        hathi.HathiItemForbidden: 'Permission denied to download data.',
+        # only saw this one on day, but this was what it was
+        JSONDecodeError: 'HathiTrust catalog temporarily unavailable.'
```

Comment:
I think this error might be worth making more specific (for when a user reports a problem to us): i.e., `HathiTrust catalog temporarily unavailable, malformed response` as a flag so that we know the JSON was bad in some way?